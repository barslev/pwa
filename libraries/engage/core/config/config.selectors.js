import { createSelector } from 'reselect';

/**
 * Retrieves the config state from the store.
 * @param {Object} state The current application state.
 * @return {Object} The locations state.
 */
function getState(state) {
  return state.config || {};
}

/**
 * Creates the selector that retrieves the config.
 * @returns {Function}
 */
export function makeGetConfig() {
  /**
   * @param {Object} state The application state.
   * @returns {Object}
   */
  return createSelector(
    getState,
    state => state
  );
}

/**
 * Creates the selector that retrieves the merchant settings.
 * @returns {Function}
 */
export function makeGetMerchantSettings() {
  /**
   * @param {Object} state The application state.
   * @returns {Object}
   */
  return createSelector(
    getState,
    state => state.merchantSettings || {}
  );
}
