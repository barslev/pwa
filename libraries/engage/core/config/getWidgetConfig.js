import { getPageConfig } from './getPageConfig';

/**
 * Retrieves the config for a specific widget by its id and an optional index on a given page.
 * If no index is given it will return the first found widget config by widgetId of that page.
 * If no widget is found on the current page or if the page does not exist, an empty object is
 * returned.
 * The returned widget config is pure and the widget settings are not inherited!
 *
 * @param {string} pagePattern The pattern of the page where the widget is located.
 * @param {string} widgetId The id of the widget to look for, which must exist in the config.
 * @param {number|undefined} [index=0] The optional index of the widget.
 * @returns {Object}
 */
export function getWidgetConfig(pagePattern, widgetId, index = 0) {
  const { widgets = [] } = getPageConfig(pagePattern, widgetId);

  const widgetConfig = [].concat(widgets).find((element, i) => {
    if (!index) {
      return element.id === widgetId;
    }

    return element.id === widgetId && i === index;
  });

  if (!widgetConfig) {
    return {};
  }

  return widgetConfig;
}
