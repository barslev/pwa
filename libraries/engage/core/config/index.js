export { default as subscriptions } from './config.subscriptions';
export { default as reducers } from './config.reducers';

export * from './config.actions';
export * from './config.selectors';
export * from './config.streams';
