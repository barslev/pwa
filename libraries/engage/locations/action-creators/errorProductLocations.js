import { ERROR_PRODUCT_LOCATIONS } from '../constants';

/**
 * Dispatches the ERROR_PRODUCT_LOCATIONS action object.
 * @param {string} productId The ID of the product.
 * @param {string} errorCode errorCode
 * @return {Object} The ERROR_PRODUCT_LOCATIONS action.
 */
const errorProductLocations = (productId, errorCode) => ({
  type: ERROR_PRODUCT_LOCATIONS,
  productId,
  errorCode,
});

export default errorProductLocations;
