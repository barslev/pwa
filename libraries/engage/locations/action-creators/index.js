export { default as selectLocation } from './selectLocation';
export { default as storeFormInput } from './storeFormInput';
export { default as errorProductLocations } from './errorProductLocations';
export { default as receiveProductLocations } from './receiveProductLocations';
export { default as requestProductLocations } from './requestProductLocations';
export { default as submitReservationRequest } from './submitReservationRequest';
export { default as submitReservationSuccess } from './submitReservationSuccess';
export { default as submitReservationError } from './submitReservationError';
