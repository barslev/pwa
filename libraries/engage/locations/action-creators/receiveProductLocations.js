import { RECEIVE_PRODUCT_LOCATIONS } from '../constants';

/**
 * Creates the dispatched RECEIVE_PRODUCT_LOCATIONS action object.
 * @param {string} productId The ID of the product.
 * @param {Array} locations List of product locations.
 * @return {Object} The RECEIVE_PRODUCT_LOCATIONS action.
 */
const receiveProductLocations = (productId, locations) => ({
  type: RECEIVE_PRODUCT_LOCATIONS,
  productId,
  locations,
});

export default receiveProductLocations;
