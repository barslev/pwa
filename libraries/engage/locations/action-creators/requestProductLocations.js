import { REQUEST_PRODUCT_LOCATIONS } from '../constants';

/**
 * Dispatches the REQUEST_PRODUCT_LOCATIONS action.
 * @param {string} productId The ID of the product to request the locations for.
 * @return {Object} The REQUEST_PRODUCT_LOCATIONS action.
 */
const requestProductLocations = productId => ({
  type: REQUEST_PRODUCT_LOCATIONS,
  productId,
});

export default requestProductLocations;
