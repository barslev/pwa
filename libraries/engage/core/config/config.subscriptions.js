import { appWillStart$ } from '@shopgate/pwa-common/streams';
import { fetchConfig } from './config.actions';

/**
 * Config subscriptions.
 * @param {Function} subscribe The subscribe function.
 */
function config(subscribe) {
  subscribe(appWillStart$, ({ dispatch }) => {
    dispatch(fetchConfig());
  });
}

export default config;
