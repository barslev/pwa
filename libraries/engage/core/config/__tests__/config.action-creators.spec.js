import { requestConfig, receiveConfig, errorConfig } from '../config.action-creators';

describe('engage > core > config > actions-creators', () => {
  test('requestConfig', () => {
    expect(requestConfig()).toMatchSnapshot();
  });

  test('receiveConfig', () => {
    expect(receiveConfig({
      merchantSettings: {
        foo: 'bar',
      },
    })).toMatchSnapshot();
  });

  test('errorConfig', () => {
    expect(errorConfig(new Error('Pipeline failed'))).toMatchSnapshot();
  });
});
