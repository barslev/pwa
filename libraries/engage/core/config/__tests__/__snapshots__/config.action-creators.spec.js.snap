// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engage > core > config > actions-creators errorConfig 1`] = `
Object {
  "error": [Error: Pipeline failed],
  "type": "ERROR_CORE_CONFIG",
}
`;

exports[`engage > core > config > actions-creators receiveConfig 1`] = `
Object {
  "config": Object {
    "merchantSettings": Object {
      "foo": "bar",
    },
  },
  "type": "RECEIVE_CORE_CONFIG",
}
`;

exports[`engage > core > config > actions-creators requestConfig 1`] = `
Object {
  "type": "REQUEST_CORE_CONFIG",
}
`;
